packages/widgetbook/lib/src/knobs/duration_knob.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:meta/meta.dart';
+
+import '../fields/fields.dart';
+import '../state/state.dart';
+import 'knob.dart';
+
+@internal
+class DurationKnob extends Knob<Duration> {
+  DurationKnob({
+    required super.label,
+    required super.value,
+    super.description,
+  });
+
```

Comment:
Consider adding `DurationKnob.nullable` constructor, and a `durationOrNull` variant to the `KnobsBuilder`. That will be a great addition that gives a lot of flexibility. (Don't forget to add a test group for that as well).